src/applications/MachineShopSimulator.java

```diff
@@ -82,13 +82,13 @@ private static void setUpJobs(SimulationSpecification specification) {
             // create the job
             theJob = new Job(i);
             for (int j = 1; j <= tasks; j++) {
-                int theMachine = specification.getJobSpecifications(i).getSpecificationsForTasks()[2*(j-1)+1];
-                int theTaskTime = specification.getJobSpecifications(i).getSpecificationsForTasks()[2*(j-1)+2];
+                int theMachine = specification.getJobSpecificationTasks(i)[2*(j-1)+1];
+                int theTaskTime = specification.getJobSpecificationTasks(i)[2*(j-1)+2];
                 if (j == 1)

```

Comment:
Nice job creating the method in `SimulationSpecification` and removing the message chain here. `[2*(j-1)+1]` could be made easier to read by starting the for loop at 0, and changing` 2*(j-1)` to `2*j`